composables/useUser/index.ts

```diff
@@ -0,0 +1,98 @@
+import { computed, reactive } from "@vue/composition-api"
+import { getCurrentUser } from "../../gql/queries"
+import { loginMutation } from "../../gql/mutations"
+import { LogInUserParams, User } from "../types"
+import { storeClientCookie, removeClientCookie } from "../helpers/cookieHelper"
+import { useState, useNuxtApp } from "#app"
+
+export const useUser = () => {
+  const nuxt = useNuxtApp()
+  const fetcher = nuxt.nuxt2Context.$gqlFetch
+  const user = useState(`use-user-user`, (): User => {
+    return {} as User
+  })
+  const authCookieName = nuxt.nuxt2Context.$config.userCookieKey
+  const isAuthenticated = useState(`use-user-isAuthenticated`, () => false)
+  const loading = useState(`use-user-loading`, () => false)
+  const error = reactive({
+    login: null,
+    register: null,
+  })
+  const resetErrorValues = () => {
+    error.login = null
+    error.register = null
+  }
+
+  const load = async () => {
+    try {
+      loading.value = true
+      const response = await fetcher({
+        query: getCurrentUser,
+        variables: {},
+      })
+      user.value = response.data?.customerAccount
+    } catch (err) {
+    } finally {
+      loading.value = false
+    }
+  }
+
+  // User
+  const login = async (params: LogInUserParams) => {
```

Comment:
LogInUserParams can just be CustomerUserAuthInfoInput

Reply:
used CustomerUserAuthInfoInput instead LogInUserParams